internal/controller/managedcluster_controller.go

```diff
@@ -514,6 +555,36 @@ func (r *ManagedClusterReconciler) machinesAvailable(ctx context.Context, namesp
 	return len(itemsList.Items) != 0, nil
 }
 
+func setIdentityHelmValues(values *apiextensionsv1.JSON, idRef *corev1.ObjectReference) (*apiextensionsv1.JSON, error) {
+	var valuesJSON map[string]any
+	err := json.Unmarshal(values.Raw, &valuesJSON)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling values: %s", err)
+	}
+
+	identityVals := map[string]any{
+		"name": idRef.Name,
+	}
+
+	switch idRef.Kind {
```

Comment:
We should not be provider-specific here. whatever is defined in the `idRef` should be passed as is. It should be up to the template owner to decide which fields to use.